src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "unittest.h"
+#include "ut_pmem2.h"
+#include "shutdown_state.h"
+#include "set.h"
+
+int os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush);
+/*
+ * os_part_deep_common -- XXX temporally workaround until we will have pmem2
+ *		integrated with common
+ */
+int
+os_part_deep_common(struct pool_replica *rep, unsigned partidx, void *addr,
+	size_t len, int flush)
+{
+	return 0;
+}
+
```

Comment:
You will need this mock and these globals as well. Sorry it complicates your tests a little bit:

```c
static struct shutdown_state Pool_sds;
static struct shutdown_state Curr_sds;
static struct pool_replica Rep;

FUNC_MOCK(shutdown_state_reinit, void, struct shutdown_state *curr_sds,
	struct shutdown_state *pool_sds, struct pool_replica *rep)
FUNC_MOCK_RUN_DEFAULT {
	ASSERTeq(curr_sds, &Curr_sds);
	ASSERTeq(pool_sds, &Pool_sds);
	ASSERTeq(rep, &Rep);
}
FUNC_MOCK_END
```

Reply:
You can use a macro like this to "set" value of globals on the spot. Assuming you find it useful:

```c
#define SET_SDS(DST, VALUE) \
{ \
	struct shutdown_state sds_src = VALUE; \
	ASSERTeq(memcmp(&DST, &sds_src, sizeof(struct shutdown_state)), 0); \
}
```

For an example of how you can use it please see the first test case.

Reply:
Done.